external/avro/src/main/scala/org/apache/spark/sql/avro/AvroUtils.scala

```diff
@@ -330,4 +332,9 @@ private[sql] object AvroUtils extends Logging {
     case Seq() => "top-level record"
     case n => s"field '${n.mkString(".")}'"
   }
+
+  /** Return true iff `avroField` is nullable, i.e. `UNION` type and has `NULL` as an option. */
```

Comment:
Why does it need to be `UNION` type ?
If it does need to be, perhaps rename `isNullable` to convey that as well ?

Reply:
Avro represents a nullable field as a union of null and the concrete type, e.g. `union<null, string>` is a nullable string in Avro parlance. A union of null and a single type is equivalent to a nullable field which is not a union, from Spark's perspective. I think `isNullable` is still the correct name; from Spark's perspective, we are checking nullability. The fact that Avro represents this as a union is an implementation detail.